libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -373,7 +370,19 @@ bool ASTJsonConverter::visit(FunctionDefinition const& _node)
 		make_pair("scope", idOrNull(_node.scope()))
 	};
 
-	if (_node.isPartOfExternalInterface())
+
+	if (_node.annotation().contract)
+	{
+		Visibility visibility;
+		if (_node.isConstructor())
+			visibility = _node.annotation().contract->abstract() ? Visibility::Internal : Visibility::Public;
+		else
+			visibility = _node.visibility();
+
+		attributes.emplace_back("visibility", Declaration::visibilityToString(visibility));
+	}
+
+	if (_node.isPartOfExternalInterface() && TypeProvider::function(_node)->interfaceFunctionType())
```

Comment:
When would the first be true but not the second?

Reply:
For example when the `parameters()` members don't have `annotation().type` information.